internal/codegen/type.go

```diff
@@ -25,6 +25,15 @@ type RestliType struct {
 	Union     *UnionType
 }
 
+func (t *RestliType) Nil() *Statement {
+	switch {
+	case t.Primitive != nil:
+		return t.Primitive.Nil()
+	default:
+		return Nil()
+	}
+}
```

Comment:
I'm worried this may bite since typerefs that point to primitives aren't usually used as pointers, just the raw value. Can you double check this works across the board?